src/lib/db-listeners/db-publications-listener.ts

```diff
@@ -54,6 +54,22 @@ export class DBPublicationsListener {
       this.db.createDocumentFromPublication(publication)
         .then(doc => {
           documents.add(doc);
+          onPatch(doc.stars, patch => {
+            const path = patch.path.split("/");
+            if (patch.op === "add") {
+              const star = patch.value;
+              const { starred, key } = star;
+              if (key === "") { // key will be assigned from Firebase node id
+                this.db.createUserStar(doc, starred);
+              }
+            } else if (patch.op === "replace" && path[path.length - 1] === "starred") {
+              const starIndex = parseInt(path[path.length - 2], 10);
```

Comment:
In reviewing @ekosmin 's commenting PR, we discussed the reliance on the internals of the patch format in this code. I made the point that since MST is using the standard JSON patch format, we're not relying on some MST-specific construct that's likely to change in future versions of MST, so it's not a worrying dependency in that sense. It does make the code rather inscrutable, however. Moving forward, perhaps a `json-patch-utils` module would make things clearer. Or perhaps a destructuring assignment of the form:
```
const [ , tileId, ..., starIndex, replaceKey ] = patch.path.split("/")
```
might be sufficient.

Reply:
I agree, unless you set up break points or log statements and actually see the patch operation values coming through, it's next to impossible to comprehend this code (I was certainly baffled by the parsing when I first looked at the comments commit).  Anyway, I don't have any big insight here, other than to say that the concerns seem valid.

Reply:
As a near-term improvement, the destructuring suggestion seems like it could be a step in the right direction.

Reply:
Agreed.  I added it for both the stars and comments onPatch cases.

Reply:
@ekosmin Could you double-check my changes in the comments `onPatch` to ensure that my destructuring looks ok?  I did a couple of quick tests, but would like a second pair of eyes to look.  Thanks!